src/test/scala/com/databricks/spark/redshift/RedshiftSuite.scala

```diff
@@ -0,0 +1,458 @@
+           |1239012341823719, 23, 'Unicode''s樂趣', '2015-07-01 00:00:00.001')
+        """.stripMargin
+      ).executeUpdate()
+
+      conn.commit()
+    }
+
+    createTable("test_table")
+    createTable("test_table2")
+    createTable("test_table3")
+
+    conn.commit()
+  }
+
+  override def afterAll(): Unit = {
+    val temp = new File(tempDir)
+    val tempFiles = temp.listFiles()
+    if(tempFiles != null) tempFiles foreach {
+      case f => if(f != null) f.delete()
+    }
+    temp.delete()
+
+    conn.prepareStatement("drop table if exists test_table").executeUpdate()
+    conn.prepareStatement("drop table if exists test_table2").executeUpdate()
+    conn.prepareStatement("drop table if exists test_table3").executeUpdate()
+    conn.commit()
+    conn.close()
+
+    sc.stop()
+    super.afterAll()
+  }
+  
+  test("DefaultSource can load Redshift UNLOAD output to a DataFrame") {
+    val sqlContext = new SQLContext(sc)
+    sqlContext.sql(
+      s"""
+         |create temporary table test_table(
+         |  testbyte tinyint,
+         |  testbool boolean,
+         |  testdate date,
+         |  testdouble double,
+         |  testfloat float,
+         |  testint int,
+         |  testlong bigint,
+         |  testshort smallint,
+         |  teststring string,
+         |  testtimestamp timestamp
+         |)
+         |using com.databricks.spark.redshift
+         |options(
+         |  url \"$jdbcUrl\",
+         |  tempdir \"$tempDir\",
+         |  dbtable \"test_table\"
+         |)
+       """.stripMargin
+    ).collect()
+
+    sqlContext.sql("select * from test_table order by testbyte, testbool").collect()
+      .zip(expectedData).foreach {
+      case (loaded, expected) => loaded shouldBe expected
+    }
+  }
+
+  test("DefaultSource supports simple column filtering") {
+    val sqlContext = new SQLContext(sc)
+    sqlContext.sql(
+      s"""
+         |create temporary table test_table(
+         |  testbyte tinyint,
+         |  testbool boolean,
+         |  testdate date,
+         |  testdouble double,
+         |  testfloat float,
+         |  testint int,
+         |  testlong bigint,
+         |  testshort smallint,
+         |  teststring string,
+         |  testtimestamp timestamp
+         |)
+         |using com.databricks.spark.redshift
+         |options(
+         |  url \"$jdbcUrl\",
+         |  tempdir \"$tempDir\",
+         |  dbtable \"test_table\"
+         |)
+       """.stripMargin
+    )
+
+    val prunedExpectedValues = Array(
+      Row(null, null),
+      Row(0.toByte, null),
+      Row(0.toByte, false),
+      Row(1.toByte, false),
+      Row(1.toByte, true)
+    )
+
+    sqlContext.sql("select testbyte, testbool from test_table order by testbyte, testbool").collect()
+      .zip(prunedExpectedValues).foreach {
+      case (loaded, expected) => loaded shouldBe expected
+    }
+  }
+
+  test("DefaultSource supports user schema, pruned and filtered scans") {
```

Comment:
I think that this test should be renamed in this suite since it's not actually asserting that a pruned or filtered scan is taking place (i.e. it's not directly testing whether pushdown actually took place). If we keep the name, then we might want to strengthen the assertion to actually assert something about the number of rows loaded from Redshift vs. the total number of rows in the DB.